### PR TITLE
Allow serving app from non-root path in dev env

### PR DESCRIPTION
**Pull request by shanemcd**
##### SUMMARY
Usage:

```
$ EXTRA_SOURCES_ANSIBLE_OPTS='-e ingress_path=/awx' make docker-compose 
$ curl http://localhost:8013/awx/api/v2/ping/
```

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other

### CODE REVIEW

File: tools/docker-compose/ansible/roles/sources/tasks/main.yml [hunk 1]
```diff
@@ -49,19 +49,17 @@
...
+    - "local_settings.py"
     - "websocket_secret.py"
     - "haproxy.cfg"
+    - "nginx.conf"
+    - "nginx.locations.conf"
 
 - name: Delete old local_settings.py
   file:
     path: "{{ playbook_dir }}/../../../awx/settings/local_settings.py"
     state: absent
```
Comment (AlanCoding) [inline tools/docker-compose/ansible/roles/sources/tasks/main.yml]: I don't think you need the "Delete old local_settings.py" task anymore either?

File: tools/docker-compose/ansible/roles/sources/templates/docker-compose.yml.j2 [hunk 2]
```diff
@@ -40,6 +41,8 @@ services:
       - "../../docker-compose/_sources/database.py:/etc/tower/conf.d/database.py"
       - "../../docker-compose/_sources/websocket_secret.py:/etc/tower/conf.d/websocket_secret.py"
       - "../../docker-compose/_sources/local_settings.py:/etc/tower/conf.d/local_settings.py"
```
Comment (AlanCoding) [inline tools/docker-compose/ansible/roles/sources/templates/docker-compose.yml.j2:43]: well this is confusing that we had 2 paths (on the host machine) where local_settings.py could be - either in `awx/settings/local_settings.py` or in `tools/docker-compose/_sources/local_settings.py`

File: tools/docker-compose/ansible/roles/sources/templates/nginx.conf.j2 [hunk 3]
```diff
@@ -17,106 +16,61 @@ http {
...
-            uwsgi_read_timeout 120s;
-            uwsgi_pass uwsgi;
-            include /etc/nginx/uwsgi_params;
-            {%- if extra_nginx_include is defined %}
-            include {{ extra_nginx_include }};
-            {%- endif %}
-            proxy_set_header X-Forwarded-Port 443;
-            uwsgi_param HTTP_X_FORWARDED_PORT 443;
-        }
+        include /etc/nginx/conf.d/*.conf;
```
Comment (AlanCoding) [inline tools/docker-compose/ansible/roles/sources/templates/nginx.conf.j2:74]: interesting approach

**Review by AlanCoding — APPROVED** [review]
great cleanup work

**Comment by akus062381** [comment]
this PR seems to have introduced several UI failures. I created a PR reverting it and I am running tests against the reverted PR to see if the failures are fixed.

**Comment by AlanCoding** [comment]
We had another reversion with https://github.com/ansible/awx/pull/13623, which reverted something known to have caused a bunch of failures. Your PR seems to have happened after that revert merged, so it might be worth getting results from the devel branch so we know it's from one or the other.

**Pull request merged**